### PR TITLE
fix: transactions tracking

### PR DESCRIPTION
**Pull request by janek26**

### CODE REVIEW

File: packages/extension/src/background/trackTransactions.ts [hunk 1]
```diff
@@ -87,7 +91,13 @@ export class TransactionTracker {
       }),
     )
     if (transactionStatus.length > 0) {
-      this.transactions = transactionStatus
+      // add transactions that were added while we were fetching
+      this.transactions = [
+        ...transactionStatus,
```
Comment (delaaxe) [inline packages/extension/src/background/trackTransactions.ts:96]: if array should be plural `transactionStatuses`

File: packages/extension/src/background/trackTransactions.ts [hunk 2]
```diff
@@ -87,7 +91,13 @@ export class TransactionTracker {
...
+        ...this.transactions.filter((tx) => {
```
Comment (delaaxe) [inline packages/extension/src/background/trackTransactions.ts:97]: is it sure this concatenation will result in unique elements?

Comment (delaaxe) [reply packages/extension/src/background/trackTransactions.ts:97]: avoid abbreviations like `tx`

**Review by delaaxe — APPROVED** [review]

File: packages/extension/src/background/trackTransactions.ts (hunk 2 above)
Comment (janek26) [reply packages/extension/src/background/trackTransactions.ts:97]: yes it does:
As a transaction hash is a unique identifier of a transaction and this operation is taking a superset of the fetched transactions and removes the intersection before merging

**Pull request merged**